From ac4f1acc01167561658013ec0ae827fc44f6950a Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 02:33:24 -0400
Subject: [iommu] Fix IOMMU sysfs stubs

Message-id: <20140905023324.497.21254.stgit@gir.stowe>
Patchwork-id: 91383
O-Subject: [RHEL7.1 PATCH 6/7] iommu: Fix IOMMU sysfs stubs
Bugzilla: 1136539
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1136539
Upstream Status: e09f8ea560490e941139d23b4c278d3e6e2c871a

commit e09f8ea560490e941139d23b4c278d3e6e2c871a
Author: Alex Williamson <alex.williamson@redhat.com>
Date:   Mon Jul 7 14:31:36 2014 -0600

    iommu: Fix IOMMU sysfs stubs

    0-day kernel build testing reports:

       arch/x86/kvm/x86.o: In function `iommu_device_destroy':
    >> (.text+0x7a0a): multiple definition of `iommu_device_destroy'
       arch/x86/kvm/../../../virt/kvm/vfio.o:vfio.c:(.text+0x490): first defined here
       arch/x86/kvm/x86.o: In function `iommu_device_link':
    >> (.text+0x7a15): multiple definition of `iommu_device_link'
       arch/x86/kvm/../../../virt/kvm/vfio.o:vfio.c:(.text+0x49b): first defined here
       arch/x86/kvm/x86.o: In function `iommu_device_unlink':
    >> (.text+0x7a25): multiple definition of `iommu_device_unlink'
       arch/x86/kvm/../../../virt/kvm/vfio.o:vfio.c:(.text+0x4ab): first defined here
       arch/x86/kvm/x86.o: In function `iommu_device_create':
    >> (.text+0x79f8): multiple definition of `iommu_device_create'
       arch/x86/kvm/../../../virt/kvm/vfio.o:vfio.c:(.text+0x47e): first defined here

    These are due to failing to define the stubs as static inline.  Fix.

    Signed-off-by: Alex Williamson <alex.williamson@redhat.com>
    Signed-off-by: Joerg Roedel <jroedel@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index 03b2469..6bab130 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -387,23 +387,24 @@ static inline int iommu_domain_set_attr(struct iommu_domain *domain,
  return -EINVAL;
 }
 
-struct device *iommu_device_create(struct device *parent, void *drvdata,
-       const struct attribute_group **groups,
-       const char *fmt, ...)
+static inline struct device *iommu_device_create(struct device *parent,
+     void *drvdata,
+     const struct attribute_group **groups,
+     const char *fmt, ...)
 {
  return ERR_PTR(-ENODEV);
 }
 
-void iommu_device_destroy(struct device *dev)
+static inline void iommu_device_destroy(struct device *dev)
 {
 }
 
-int iommu_device_link(struct device *dev, struct device *link)
+static inline int iommu_device_link(struct device *dev, struct device *link)
 {
  return -EINVAL;
 }
 
-void iommu_device_unlink(struct device *dev, struct device *link)
+static inline void iommu_device_unlink(struct device *dev, struct device *link)
 {
 }
 
-- 
1.7.1